### DISCUSSION TITLE
Space between `:` and label, key and `:`?

### DISCUSSION THREAD

**Discussion by nichtich** [pg-format/pg-formatter#25] [Q&A] [answered]
Only looking at the PG grammar I found out that space is allowed between colon and label and between property key and colon, so these are equivalent (apart from node id):

~~~
1 :event year:2024
2 :event year : 2024
3 : event year: 2024
4 : event year : 2024
~~~

I'd prefer only 1 to be allowed. 2 may be useful too, but the space between colon and label makes it less readable. The reading rule "if it starts with a colon, it must be a label" should apply. 

By the way if space is allowed, how about this?

~~~
5 :
  event
  year
  :
  2024
~~~

It would be easier not to allow the space at all. However this *may* be useful:

~~~
6 :event
  year
    :
      2024,
      2025
~~~

**Comment by hchiba1** [comment, marked as answer]
I'd like to allow space between `:` and values (Cypher style). I don't mind even if they are on different lines.
However, I think that the key and `:` should be more closely associated, i.e., on the same line.
```
7 :event
  year:
      2024,
      2025
```
I also think `:` and the label should be close (on the same line), still allowing space.

That is , in EBNF,
```ebnf
Label     ::= WS ':' SPACE_OR_TAB* String
Property  ::= WS String SPACE_OR_TAB* ':' WS? ValueList
```

**Reply by nichtich** [reply]
Ok, I've check with Cypher syntax, that's a good argument. The formatter should remove the spaces, though. Having them on separate lines, however would look rather ugly and make parsing difficult. In this example `Databases` looks like a label but it is a value:

~~~
a :Book title # comment
  :Databases
~~~

So simple space and tab as implemented now should be enough. I've added this to the grammar used in pgraph!
